### ISSUE TITLE
Shaderset compilation is buggy

### ISSUE THREAD

**Issue by gamemachine** [Moonlit-Games/Ultimate-Water-System#8]
This was an issue with Playway also but appears to be worse in UWS.

Shaders in shader sets are being removed at odd times, not as documented or expected.  Selecting the water does not force a check.  However saving a scene or switching scenes can cause shader variants to get removed, but no new ones are compiled.  Most of the time I have to make sure the scene is first saved, then go to the shader set and toggle one of the missing shaders like the surface shader, then that prompts the regenerate button to show and I can regenerate.

I also find shader variants being removed with no changes to the water or shader set happening.

I think a less magical approach would work better.  Provide hints that a change requires a rebuld, but let me trigger the actual rebuild.

**Comment by MCPGNZ** [comment]
If i understand correctly, you lost the shaders from the shader-sets only after you made changes to shaders?

And yeah, you're right, i thought exactly that when i saw the design of this part of the package but it would be very time consuming to just trash it and start from scratch and decided to stay with the solution (but i'm pretty sure that i'll have do it eventually). 

questions for future me:
- [ ] is it sane to try fixing the existing solution or just reimplement/simplify it

**Comment by MCPGNZ** [comment]
Closed - duplicate of #10
